Adding the new  Steel Spinning Roller Coaster Track

Realsteel89 opened the pull request:
**Adding the new Coaster Track Maurer/Gerstlauer Spinner**
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/410bf66d-8bd7-4f93-9374-8287ecee2975)
The Track is Based on the Maurer Spinning Coaster in Real Life and based as the MiniRollerCoaster from RCT2

**What I did:**
- Building the new Track Model and render most of the sprites there are possible to the Coaster Track
- limited it with the cheat function that only the track parts are possible without cheats that the real coaster has too
- changing Supports from fork to tubes
- changing some hights from supports to look nicer
- adding vertical lift animation sprites to the coaster too (manually edding)
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/af012565-63f8-4d5d-8a6b-7a247287371e)

- adding an preview Image to coaster Type
- changing the rendering sprites from small vertical loops because had some cutes by the border of the tiles so i edding it   manually to look correct


**Mistakes that i need to fix now:**

- Fixing the small loop sprites and bounding boxes (they currently glitch badly, although removing the small loops entirely is also an option)
- fix maybe some hights of support

_this track type can be so useful for more Coaster Types like_ 
**Gerstlauer Eurofighter**
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/0c7eee17-be90-459f-9c81-14690e560e4e)
**Gerstlauer Spinner**
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/9ba2f281-85b2-40db-9713-c03cd9813bf2)
**Vekoma Family Coaster**
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/7a759aa1-8a43-4388-96ea-27b49f580810)

and so many one more....

It has 1274 Sprites so most of the Elements that the game can do at the moment limited by cheat setting on/off



_**later plans:**_
-maybe realistic spinning maurer car for the track
- vertical tower lift and an 110° Drop sprite
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/0e42baf2-9381-4928-bf65-b59253748bff)
- maybe build animated breaks like the Minicoaster has with open/close animation

**Important:** **The Problem that it exist actually no Car for the Track, you need to choose it via cheat funktion!**


I would appreciate if someone could test to ensure that the branch is functional and I didn't forget to commit any sprites.

ZehMatt commented:
Please don't close PRs and open new ones, you can always overwrite the entire git history of your branch, for now its fine but in the future try to avoid that.

fidwell commented on data/language/en-GB.txt:
```diff
@@ -98,6 +98,7 @@ STR_0093    :Hybrid Coaster
 STR_0094    :Single Rail Roller Coaster
 STR_0095    :Alpine Coaster
 STR_0096    :Classic Wooden Roller Coaster
+STR_0097    :Maurer Steel Roller Coaster
```
We need a generic, non-branded name for this, right? I think the dev team discussed it a long time ago, but I don't remember if something was picked for this. If not, "Mini Looping Roller Coaster" is my nomination.

X123M3-256 replied:
If it's meant to be the Maurer track, the obvious name is "Spinning Roller Coaster"

fidwell replied:
That makes sense!

Acesurfer replied:
Due to the potential uses as a Eurofighter or Vekoma, would "Compact Looping Roller Coaster" be a better name?  This would allow the track to be more flexible and would not lead users into believing it is purely a spinning coaster.

Realsteel89 replied:
the renaming later was no problem i thinked of this at least too.
the Inversions are only enable as you use cheats for maurer only there is most elements blocked without cheats.

**SET_FIELD(EnabledTrackPieces**, { TRACK_FLAT, TRACK_STRAIGHT, TRACK_STATION_END, TRACK_LIFT_HILL, TRACK_LIFT_HILL_STEEP, TRACK_FLAT_ROLL_BANKING, TRACK_SLOPE, TRACK_SLOPE_LONG, TRACK_SLOPE_STEEP_UP, TRACK_SLOPE_STEEP_DOWN, TRACK_SLOPE_CURVE, TRACK_SLOPE_CURVE_STEEP, TRACK_S_BEND, TRACK_CURVE_SMALL, TRACK_CURVE, TRACK_CURVE_LARGE, TRACK_HELIX_DOWN_BANKED_HALF, TRACK_HELIX_UP_BANKED_HALF, TRACK_BRAKES, TRACK_ON_RIDE_PHOTO, TRACK_SLOPE_VERTICAL, TRACK_SLOPE_CURVE_BANKED, TRACK_BLOCK_BRAKES, TRACK_SLOPE_ROLL_BANKING, TRACK_CURVE_VERTICAL, TRACK_SLOPE_CURVE_LARGE, TRACK_POWERED_LIFT}),
    
**SET_FIELD(ExtraTrackPieces,**  {TRACK_BOOSTER, TRACK_SLOPE_STEEP_LONG, TRACK_CORKSCREW, TRACK_BARREL_ROLL, TRACK_BARREL_ROLL, TRACK_QUARTER_LOOP, TRACK_HALF_LOOP, TRACK_HALF_LOOP_MEDIUM, TRACK_HALF_LOOP_LARGE, TRACK_VERTICAL_LOOP, TRACK_CORKSCREW_LARGE, TRACK_ZERO_G_ROLL, TRACK_ZERO_G_ROLL_LARGE}),
    SET_FIELD(CoveredTrackPieces, {}),

Realsteel89 replied:
> 

maybe if you want i had your pinfari model and can do it later too

Acesurfer replied:
Makes sense.  I was unsure if renaming at a later date was a possibility.  I appreciate the information.

X123M3-256 replied:
> Due to the potential uses as a Eurofighter or Vekoma, would "Compact Looping Roller Coaster" be a better name? This would allow the track to be more flexible and would not lead users into believing it is purely a spinning coaster.

Well you have to decide what ride type it's supposed to be. If you want to use the same track type for multiple ride types, there should really be multiple RTDs with different names that just share track paint code.

The name should match how the ride is set up - if it's supposed to be a spinning coaster, the inversions should not be available without cheats, the stats, max height, build cost etc should all reflect the spinning coaster. If it's supposed to be a Eurofighter that stuff should be different.

I really don't think this track should be labelled a Eurofighter if it doesn't have the beyond vertical drop (I'm also not sure that this track best fits a Eurofighter, only the very earliest examples had this style). In terms of being a ride that isn't already in the game, and really should be, I think the Maurer/Gerstlauer spinner is a good choice.

Realsteel89 replied:
> > Due to the potential uses as a Eurofighter or Vekoma, would "Compact Looping Roller Coaster" be a better name? This would allow the track to be more flexible and would not lead users into believing it is purely a spinning coaster.
> 
> Well you have to decide what ride type it's supposed to be. If you want to use the same track type for multiple ride types, there should really be multiple RTDs with different names that just share track paint code.
> 
> The name should match how the ride is set up - if it's supposed to be a spinning coaster, the inversions should not be available without cheats, the stats, max height, build cost etc should all reflect the spinning coaster. If it's supposed to be a Eurofighter that stuff should be different.
> 
> I really don't think this track should be labelled a Eurofighter if it doesn't have the beyond vertical drop (I'm also not sure that this track best fits a Eurofighter, only the very earliest examples had this style). In terms of being a ride that isn't already in the game, and really should be, I think the Maurer/Gerstlauer spinner is a good choice.



as said the inversions are all not available in normal mode. "Maurer/Gerstlauer Roller Coaster" is not bad. It was planned as Maurer Sc3000 I just meant you could use it as one of the first models for gerstlauer eurofighter but you don't have to.

I will call it "Maurer/Gerstlauer Roller Coaster".

X123M3-256 replied:
> I will call it "Maurer/Gerstlauer Spinner".

I recommend just calling it "Spinning Roller Coaster" to be consistent with the game's naming convention. None of the vanilla rides mention the manufacturer name - and it isn't really needed, because the only other spinning coaster is called "spinning wild mouse" which is different enough.

Also, not having the manufacturer listed means that it could have both Maurer and Gerstlauer trains as options, which is consistent with the way rides like the hyper twister work - these don't really need to be separate ride types, just different trains as they are so similar.

duncanspumpkin replied:
Agreed do not mention manufacturer names it should be generic.

X123M3-256 commented on src/openrct2/ride/coaster/meta/MaurerRollerCoaster.h:
```diff
@@ -0,0 +1,86 @@
+    SET_FIELD(ColourKey, RideColourKey::Ride),
+    SET_FIELD(Name, "trex_rc"),
+
+    SET_FIELD(RatingsData,
+    {
+        RatingsCalculationType::Normal,
+        { RIDE_RATING(2, 55), RIDE_RATING(2, 40), RIDE_RATING(1, 85) },
+        13,
+        -1,
+        false,
+        {
+            { RatingsModifierType::BonusLength,           6000,             764, 0, 0 },
+            { RatingsModifierType::BonusSynchronisation,  0,                RIDE_RATING(0, 40), RIDE_RATING(0, 05), 0 },
+            { RatingsModifierType::BonusTrainLength,      0,                187245, 0, 0 },
+            { RatingsModifierType::BonusMaxSpeed,         0,                44281, 88562, 35424 },
+            { RatingsModifierType::BonusAverageSpeed,     0,                291271, 436906, 0 },
+            { RatingsModifierType::BonusDuration,         150,              26214, 0, 0 },
+            { RatingsModifierType::BonusGForces,          0,                20480, 23831, 49648 },
+            { RatingsModifierType::BonusTurns,            0,                26749, 34767, 45749 },
+            { RatingsModifierType::BonusDrops,            0,                29127, 46811, 49152 },
+            { RatingsModifierType::BonusSheltered,        0,                25700, 30583, 35108 },
+            { RatingsModifierType::BonusReversedTrains,   0,                2, 10, 12 },
+            { RatingsModifierType::BonusProximity,        0,                20130, 0, 0 },
+            { RatingsModifierType::BonusScenery,          0,                9760, 0, 0 },
+            { RatingsModifierType::RequirementDropHeight, 12,               2, 2, 2 },
+            { RatingsModifierType::RequirementMaxSpeed,   0x70000,          2, 2, 2 },
+            { RatingsModifierType::RequirementNegativeGs, FIXED_2DP(0, 50), 2, 2, 2 },
+            { RatingsModifierType::RequirementNumDrops,   2,                2, 2, 2 },
+            { RatingsModifierType::PenaltyLateralGs,      0,                20480, 23831, 49648 },
```
These values appear to have been copied unchanged from the mini coaster, which is a good starting point but maybe it should be changed a bit? If it's meant to be a spinning coaster, for example, perhaps the base excitement and nausea values could go up a bit, or the bonus for turns could increase.

I usually look at multiple existing rides that have similarities, here, I'd look at the mini coaster but also the spinning wild mouse, and compare the spinning wild mouse to the regular wild mouse, and perhaps also look at the spiral coaster. I usually try to come up with ride rating values that are consistent with the existing rides without being a direct copy of any of them.  However, you will find that a lot of the values will not need to change as they're the same for almost everything.

Realsteel89 commented:
Now I changed the Name to Maurer/Gerstlauer Spinning Roller Coaster.

I changed all Sprites, Folders, Strings, .ccp and .h files to the new name that fits better

change some Valiues of the Coaster to fit better to Maurer Spinner

Realsteel89 commented:
since i'm not that deep in git i had to do a hard reset to change the comit and then rebuild everything but it seems to work now.
I have adjusted the names to most of the data so that it fits with the mapping.
Only are the variables that I would still have to revise

duncanspumpkin commented on src/openrct2/ride/RideData.cpp:
```diff
@@ -49,6 +49,7 @@
 #include "coaster/meta/LIMLaunchedRollerCoaster.h"
 #include "coaster/meta/LayDownRollerCoaster.h"
 #include "coaster/meta/LoopingRollerCoaster.h"
+#include "coaster/meta/Maurer-Gerstlauer Spinner RollerCoaster.h"
```
We dont use spaces or hyphens in file names.

Realsteel89 replied:
ok so i can change it to MaurerGerstlauuerSpinnerRollercoaster.h

duncanspumpkin replied:
Yeah don't use a manufacturer name in general. You'll see all the other coaster names are generic.

Realsteel89 replied:
yes true maybe because of the copyright rights also i will change that

SpartanFrederic104 replied:
Yeah it should be **SteelSpinningRollerCoaster**

Realsteel89 replied:
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/92539dc9-3404-4584-be5a-f2351b0c7f8c)
Now I changed all the names now its generic :)

finally we have the name now

Realsteel89 commented on data/language/en-GB.txt:
```diff
@@ -181,6 +182,7 @@ STR_0603    :A wooden style roller coaster with a steel track, allowing for stee
 STR_0604    :Riders ride single file on a narrow monorail track, as they race through tight inversions and direction changes
 STR_0605    :Riders toboggan down a meandering steel track, braking to control their speed
 STR_0606    :An older-style wooden roller coaster with a fast and rough ride, with plenty of air-time, some lateral G’s, and designed to feel ‘out-of-control’
+STR_0607    :An spinning rollercoaster on an steel track with small poles inside the track
```
> ```suggestion
> STR_0607    :A smooth, compact roller coaster on a steel track, where riders ride in spinning, individual cars
> ```

Good part I will change this if I am back on vacation Tuesday afternoon.

Realsteel89 commented:
holy fuck what i did i think i changed the branch name but i deleted it oO

Realsteel89 commented:
so now i am change some sprites, rework the animation chain of vertical peaces to look nicer, fix supports. I think i am done with it. its ready for view and checking

karst commented on distribution/changelog.txt:
```diff
@@ -1,5 +1,6 @@
 0.4.6 (in development)
 ------------------------------------------------------------------------
+- add: [#20587] Add Steel Spinning Roller Coaster Track
```
Make sure to keep to the standard used everywhere else in the changelog. Don't forget about punctuation either!

Realsteel89 replied:
I will change this how does the standard looks like?

karst replied:
Please refer to other ride type additions in the changelog. "- Feature [#20597] Add Steel Spinning Roller Coaster ride type."

karst commented on data/language/en-GB.txt:
```diff
@@ -181,6 +182,7 @@ STR_0603    :A wooden style roller coaster with a steel track, allowing for stee
 STR_0604    :Riders ride single file on a narrow monorail track, as they race through tight inversions and direction changes
 STR_0605    :Riders toboggan down a meandering steel track, braking to control their speed
 STR_0606    :An older-style wooden roller coaster with a fast and rough ride, with plenty of air-time, some lateral G’s, and designed to feel ‘out-of-control’
+STR_0607    :A smooth, compact roller coaster on a steel track, where riders ride in spinning, individual cars
```
This seems more like a description for a vehicle rather than a track/ride type.

Realsteel89 replied:
I will change this

Basssiiie commented on resources/g2/track/steelspinning/steep_to_vertical_up_lift_2.png:
Lift is invisible from the back?

Basssiiie commented on resources/g2/track/steelspinning/vertical_to_steep_up_lift_1.png:
The chain lift curve seems to be a bit off at the bottom? It starts to angle towards steep later.

Basssiiie commented on resources/g2/track/steelspinning/gentle_diag_lift_1.png:
The chain lift seems to be a bit floating here?

Realsteel89 commented:
I will rework the backface and the chain parts of the track to look nicer so i need a little work to fix it all so please no merge yet

karst commented:
> I will rework the backface and the chain parts of the track to look nicer so i need a little work to fix it all so please no merge yet

When you don't want something merged yet you should mark it as a draft until it is :)

ZehMatt commented:
What's the status on this PR?

Gymnasiast commented:
Realsteel ran into trouble with rebasing/merging/updating, and had some stuff in the changes list that shouldn’t have been there. I offered to rebase his branch (an offer he gladly took).

I don’t know if it is fully finished yet, I imagine he’ll have to go through the line notes in any case (either addressing them or marking them as solved).

ZehMatt commented:
> Realsteel ran into trouble with rebasing/merging/updating, and had some stuff in the changes list that shouldn’t have been there. I offered to rebase his branch (an offer he gladly took).

Yeah I noticed that, appreciated. 

> I don’t know if it is fully finished yet, I imagine he’ll have to go through the line notes in any case (either addressing them or marking them as solved).

Fair enough, just wondering since it's still in draft and the activity stopped.

Realsteel89 commented:
The draft comes because we renewed all sprites to fit perfect to other tracks like giga mini Co. I add now the diagonal brakes and after your big changes into the code I try to rebase it and do something completely wrong. So now gymasiast fix my branch and I can go on it to finish this for merging. Think I need to check some supports that are not really correct but most of the stuff is done and work.

Realsteel89 commented:
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/bcd0986f-0d7c-45f5-a502-4ba29640790b)
changing booster and powert lifts to an wheel looking one because fits better to the track than the lsm one

karst commented:
The track looks very nice. I will give this a spin, pun intended, later today.

karst reviewed (changes requested):
![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/d5c855df-4c44-4939-8a21-6807b19f4f5f)

This track type is very good looking.

Few points of improvement:
- Supports on large half loops
- Supports on medium half loops
- Supports on the small vertical loop
- Booster speed is too high (it's currently 107kmh, I think that could be halved for this this ride type, and then still be overkill)
- Slight boundingbox glitching with trains that can use vertical pieces, and fully glitched vehicles when they cannot draw steep pieces. See picture below

![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/999a7dfe-eacc-4e5c-ba5a-00fd0678e1cd)


I have noticed no further oddities with bounding boxes, which is very good.

On the bottom I made a somewhat realistic looking Maurer layout, and this is the result.
![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/a6e6f3ca-3301-4a7b-be2e-9ebef29d3c65)

I think the ride ratings are fine. 

I would change the lift sound to use `OpenRCT2::Audio::SoundId::LiftWildMouse` as this seems more in line with the sound you would find on something like for example "The Spider" at Lagoon.

I think it's strange that there's no sprite for spinning control. I would have kind of expected it. As you can see in the POV of Spider there's atleast a reset at the end. I don't know if the embeded piece in the track at the top of the lift after the turn is a magnet to turn on spin or not. Though I don't know the model too well so it might make sense.

Hopefully this PR can be followed up by someone allowing spinning vehicles to do inversions etc.

karst commented:
I also need to give my appreciation for the seamless connection to the mini/giga 
![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/03dce2a0-b82f-414f-8f80-c3fddd58de02)

Realsteel89 commented:
thanks for testing i will look today after work on this again and try to fix it

Realsteel89 commented:
> ![image](https://private-user-images.githubusercontent.com/1516893/291748659-d5c855df-4c44-4939-8a21-6807b19f4f5f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNzMxMzMsIm5iZiI6MTcwMzE3MjgzMywicGF0aCI6Ii8xNTE2ODkzLzI5MTc0ODY1OS1kNWM4NTVkZi00YzQ0LTQ5MzktOGEyMS02ODA3YjE5ZjRmNWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjFUMTUzMzUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTdlMjNkMmQ2OGZjYmFlZjFhMzM2ZTljNTYyNjUxYzBmMjBmMDUzYTY2NTBkOWY3MDQ3MGEwYTZlNmMwNmEzNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.UAP9CGfsLpnIEcPez32Rb3wPvpwVsp2Z7h4EC4u7Um4)
> 
> This track type is very good looking.
> 
> Few points of improvement:
> 
>     * Supports on large half loops
> 
>     * Supports on medium half loops
> 
>     * Supports on the small vertical loop
> 
>     * Booster speed is too high (it's currently 107kmh, I think that could be halved for this this ride type, and then still be overkill)
> 
>     * Slight boundingbox glitching with trains that can use vertical pieces, and fully glitched vehicles when they cannot draw steep pieces. See picture below
> 
> 
> ![image](https://private-user-images.githubusercontent.com/1516893/291752396-999a7dfe-eacc-4e5c-ba5a-00fd0678e1cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNzMxMzMsIm5iZiI6MTcwMzE3MjgzMywicGF0aCI6Ii8xNTE2ODkzLzI5MTc1MjM5Ni05OTlhN2RmZS1lYWNjLTRlNWMtYmE1YS0wMGZkMDY3OGUxY2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjFUMTUzMzUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDYwYmZjZDQ2MzNmMmY4Yzg2NzNkMTk0OWVlNGIwNzZlMzY0MzUyOWJmMTZjMjJhZTFjOTZiYjJiZTNjZmJhNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.joTOBhYLyuTB9kyQ5Fz9xPDAmccjBBep1KrfVw9jTIA)
> 
> I have noticed no further oddities with bounding boxes, which is very good.
> 
> On the bottom I made a somewhat realistic looking Maurer layout, and this is the result. ![image](https://private-user-images.githubusercontent.com/1516893/291752556-a6e6f3ca-3301-4a7b-be2e-9ebef29d3c65.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNzMxMzMsIm5iZiI6MTcwMzE3MjgzMywicGF0aCI6Ii8xNTE2ODkzLzI5MTc1MjU1Ni1hNmU2ZjNjYS0zMzAxLTRhN2ItYmUyZS05ZWJlZjI5ZDNjNjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjFUMTUzMzUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzExYWZhNjZjMjE3YmJlNzU4YWE5MjU4OGE2ZWUxODI1ZTg5ZWRmMzMwYTcwYzg2NGQ1YzVmYjM2NTQ1Y2FiMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ckcxOPY8jVpbS0_gbCt-lo9DHBCsvlsqpAhtz1WetGQ)
> 
> I think the ride ratings are fine.
> 
> I would change the lift sound to use `OpenRCT2::Audio::SoundId::LiftWildMouse` as this seems more in line with the sound you would find on something like for example "The Spider" at Lagoon.
> 
> I think it's strange that there's no sprite for spinning control. I would have kind of expected it. As you can see in the POV of Spider there's atleast a reset at the end. I don't know if the embeded piece in the track at the top of the lift after the turn is a magnet to turn on spin or not. Though I don't know the model too well so it might make sense.
> 
> Hopefully this PR can be followed up by someone allowing spinning vehicles to do inversions etc.

![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/1b8eaf25-9b8b-4850-8a77-1c471000b10f)
fixed this now

Realsteel89 commented:
fixed this bounding box problem now
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/9d5b3cf1-dc0b-44db-aa3d-b703a304d342)

Realsteel89 commented:
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/cdd1769b-7983-47d8-a435-fd8076c91628)
take some crazy layout to it

Realsteel89 commented:
The smallest 1 tile curve I will add later after the ui stretch out from gymnasiast  was merged.

Realsteel89 commented:
Adding the 1 Tile Curve to the track peaces because very usefull only the thing you need to build it is before the ui changes come a bit stupid see in the video.
https://gyazo.com/8ed48873bba5ddb9335dca6c42e216f0

fidwell commented:
Keep open please!

Realsteel89 commented:
that auto closing thing is sad :D

karst commented:
![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/589a5b5c-10fc-420b-b750-64866165c965)

Regular turns are not available for selection without cheats atm.

![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/36a6604f-195a-4b35-9e97-8a55dfa4e4df)
Supports aren't quite there yet, especially for steep pieces perhaps you can steal some code from the mini?

The supports are hiding behind track instead of being connected to them
The small flat to steep piece doesn't connect at all.

![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/d2ec2fec-cb6d-4b84-9d22-74a7d62d907a)
I don't think this looks very nice, I think they should be placed differently.

![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/0a7d5095-f909-470f-84fb-41f53bff92e2)
The supports here don't look so nice yet either, especially the one on the bottom right and the middle one.

![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/baf02a8f-5055-4c61-8196-f7d527d4da92)

This support has a hole in it.

![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/8efb5532-ae9b-4fa6-a6e6-4ae74d42a4ac)
One of these supports is floating and not supporting the track.

![image](https://github.com/OpenRCT2/OpenRCT2/assets/1516893/7bf0fcfd-35ab-4ad9-912b-78d04a240798)
The loops aren't done yet.

Please work a bit more on the supports.

You can use Spacek's [TrackZoo](https://github.com/OpenRCT2/OpenRCT2/files/14544069/TrackZoo.1.zip) to check out all supports.

Realsteel89 commented:
> ![image](https://private-user-images.githubusercontent.com/1516893/311390028-589a5b5c-10fc-420b-b750-64866165c965.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5NDk2NTUsIm5iZiI6MTcwOTk0OTM1NSwicGF0aCI6Ii8xNTE2ODkzLzMxMTM5MDAyOC01ODlhNWI1Yy0xMGZjLTQyMGItYjc1MC02NDg2NjE2NWM5NjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDlUMDE1NTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGYwYzg0YTkwMmVmMTE4YjQyZTRiZWI4MjAxODAwNDBkNzllNzU5NDViNTgyMmMwZDVmODE3ZTQ0Njg5YTg4OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.7FwJ9PZxhCXxByNv_DUgTver3tjC7B1-DWj3OaozleQ)
> 
> Regular turns are not available for selection without cheats atm.
> 
> ![image](https://private-user-images.githubusercontent.com/1516893/311390101-36a6604f-195a-4b35-9e97-8a55dfa4e4df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5NDk2NTUsIm5iZiI6MTcwOTk0OTM1NSwicGF0aCI6Ii8xNTE2ODkzLzMxMTM5MDEwMS0zNmE2NjA0Zi0xOTVhLTRiMzUtOWU5Ny04YTU1ZGZhNGU0ZGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDlUMDE1NTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTUwN2Y0YzUyNmRkNzA2ZTc5ZTQzNzljMjI4MDMyOWM2OTEyMzdiYmNhMWYwM2MwOGU2NjJhZTRkOTJlZTA0YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.GyXv9rppW9DLC9ZfukqA_6oRhK4DDv0Ijxu3nj_7FNY) Supports aren't quite there yet, especially for steep pieces perhaps you can steal some code from the mini?
> 
> The supports are hiding behind track instead of being connected to them The small flat to steep piece doesn't connect at all.
> 
> ![image](https://private-user-images.githubusercontent.com/1516893/311390286-d2ec2fec-cb6d-4b84-9d22-74a7d62d907a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5NDk2NTUsIm5iZiI6MTcwOTk0OTM1NSwicGF0aCI6Ii8xNTE2ODkzLzMxMTM5MDI4Ni1kMmVjMmZlYy1jYjZkLTRiODQtOWQyMi03NGE3ZDYyZDkwN2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDlUMDE1NTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWUxZjYzMWQ3NmY0MDk1MTI3ZDdlMzhmZjVmOWIzZGUxNzYzOTU2MThiODU4MGQ0OWY2NjViODY2YzM5MjRmOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.x4b7MO_0iBhbFoXgmb28MTNTm_GbKUCmuBSuwjJYjLA) I don't think this looks very nice, I think they should be placed differently.
> 
> ![image](https://private-user-images.githubusercontent.com/1516893/311390379-0a7d5095-f909-470f-84fb-41f53bff92e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5NDk2NTUsIm5iZiI6MTcwOTk0OTM1NSwicGF0aCI6Ii8xNTE2ODkzLzMxMTM5MDM3OS0wYTdkNTA5NS1mOTA5LTQ3MGYtODRmYi00MWY1M2JmZjkyZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDlUMDE1NTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTc2NDFmYmU1OTlmYmQ4ZmU3ODQ5OTU2NGE4YWE1MDFhOTM5YjIxMTU0OTk2YTVjYjU1Yzk2N2RiNDA3YWM2ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.KwURYvl9N0Pq2iT4aQbLS4ruDdl3rcBOg669nsbkjM0) The supports here don't look so nice yet either, especially the one on the bottom right and the middle one.
> 
> ![image](https://private-user-images.githubusercontent.com/1516893/311390461-baf02a8f-5055-4c61-8196-f7d527d4da92.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5NDk2NTUsIm5iZiI6MTcwOTk0OTM1NSwicGF0aCI6Ii8xNTE2ODkzLzMxMTM5MDQ2MS1iYWYwMmE4Zi01MDU1LTRjNjEtODE5Ni1mN2Q1MjdkNGRhOTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDlUMDE1NTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjBmNDI5YTU5NGIwNmFkMzZmMTdhNDk3ZWNlNmIwM2E5OWQ2NDQxNDMwNzMzZmNhYTlmNTJlZGRiMzYxN2YxYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.T5M6NIrKIS2AFAUg7_Zy4CfrhGQif8yUwwnLnDITo9g)
> 
> This support has a hole in it.
> 
> ![image](https://private-user-images.githubusercontent.com/1516893/311390502-8efb5532-ae9b-4fa6-a6e6-4ae74d42a4ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5NDk2NTUsIm5iZiI6MTcwOTk0OTM1NSwicGF0aCI6Ii8xNTE2ODkzLzMxMTM5MDUwMi04ZWZiNTUzMi1hZTliLTRmYTYtYTZlNi00YWU3NGQ0MmE0YWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDlUMDE1NTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWE1ZDhkYjAyMjJjMWJjNTEzNGNmZjM1YmQyNDJmZjRlMDE5YmJlYWZhOGQ3Mjg5ZWRkNDVlMDljYWE0ZWMzZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.iwlQNc92HqSl8gLF71ie9R2USAuxQWjx-mO3jHUDnz0) One of these supports is floating and not supporting the track.
> 
> ![image](https://private-user-images.githubusercontent.com/1516893/311390665-7bf0fcfd-35ab-4ad9-912b-78d04a240798.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5NDk2NTUsIm5iZiI6MTcwOTk0OTM1NSwicGF0aCI6Ii8xNTE2ODkzLzMxMTM5MDY2NS03YmYwZmNmZC0zNWFiLTRhZDktOTEyYi03OGQwNGEyNDA3OTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDlUMDE1NTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWQwNjFlMTUxYmNiZGUzZjhhYjc4Y2I5NmViNWViYzBjMmEzYWUxMDg0M2ZhYzVjYjUzZTcwYThiZWQwYmY3MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.vt8Qox9n2ZHtcZ0Y8OIH4oWqimom5K36tAmNfTbza0A) The loops aren't done yet.
> 
> Please work a bit more on the supports.
> 
> You can use Spacek's [TrackZoo](https://github.com/OpenRCT2/OpenRCT2/files/14544069/TrackZoo.1.zip) to check out all supports.

Thanks for the review. I took another look at the code and realized that the same problems occur on vanilla tracks. Thanks to the perspective, since my track runs on z =9 which is relatively high, it just seems more extreme. Supports glitch in the same places on vanilla tracks as on my track. this seems to be a general codeerror in the support code not on the track itself. with the half loops i would have to increase the supports 2 units then they glitch like on all other vanilla tracks but are closer to the track.
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/15e4cc81-6ee0-48a9-9362-109c29275ddd)
![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/e1f8c384-d24d-40f9-bec2-483e6b73ea42)

![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/7034313b-226e-4b37-baec-b53cca62278d)




the 1 tile curve has the same problems as the wild mouse steel roller coaster they also look like shit in vanilla.


![grafik](https://github.com/OpenRCT2/OpenRCT2/assets/77458610/e62e51f7-b20d-41a1-8863-e7d298043c36)



the bug with the track i have to see if i ban it from the normal mode and only make it unlockable by cheat if that works or if it's because the logic of the game can't handle 4 different tracks totz that we have already extended the ui.

Realsteel89 commented:
that pr is now ready to check some supports can't go better because the limits of the tile. That track has no car jet because we need an new animation mod for it but we can add it later than

Andrewius72cat42 reviewed (approved):
Dear Realsteel: "12 out of 10, This track is amazing and I like how versatile it is. It can be used for big coasters, small coasters, eurofighters, wild mice, and all sorts of things. I enjoyed playing it's beta version very much. I can't wait for it's release." - Andrewius.

Realsteel89 commented:
because the ui bug was fixed i updated the branch to the newest version of develop

Gymnasiast commented:
Rebased the branch again to account for recent changes (introduction of Classic Stand-Up, changes to sprites.json and the segment refactor).

Realsteel89 commented:
Thanks my friend I think this branch is ready for testing and than merging

Realsteel89 commented:
Updated to actually version

Realsteel89 commented:
updated now again

karst commented:
How does this track look attached to Intamin's three track sprites (flat zigzag, triangle and square spine)

I found out yesterday this track style is often used on family coaster together with flat zigzag and sometimes even triangle style. To reduce costs.

fidwell commented:
It looks spot on to me, at least for these simple cases:
![image](https://github.com/user-attachments/assets/7c114a06-6384-44a7-83e4-69740ec76834)
![image](https://github.com/user-attachments/assets/47c057e0-eeab-4829-a9c9-82da4dd11b7d)
![image](https://github.com/user-attachments/assets/b37c20cc-d07b-414e-9f4c-9feb42a1057a)

Realsteel89 commented:
Some tubes can't connect near the track but I think that is an vanilla problem. The track was conzept by fitting 100% in giga and mini so you can mix it with this tracks perfect.

Realsteel89 commented:
![grafik](https://github.com/user-attachments/assets/f3f56243-a387-40da-ae2f-a9636294ab90)
![grafik](https://github.com/user-attachments/assets/42a24cd4-4de2-42fd-a422-f061a751581e)
![grafik](https://github.com/user-attachments/assets/6eced182-e3c3-47b9-a652-3883b7ee1912)
![grafik](https://github.com/user-attachments/assets/57f5c800-e773-4cd2-a77a-e1d9420557a3)
![grafik](https://github.com/user-attachments/assets/7b57be9e-9d59-4c93-a0b6-3ddbaf96c47a)
![grafik](https://github.com/user-attachments/assets/9a8b59ff-6ff6-4ccd-8a95-61be546ef7da)
![grafik](https://github.com/user-attachments/assets/bbcc3200-4ee6-4fd8-93ca-98d1542866d9)
![grafik](https://github.com/user-attachments/assets/5b3bcda7-2a13-4286-86ca-556613ed5d3d)
![grafik](https://github.com/user-attachments/assets/07793da0-5849-4c9e-b300-24a415e41cd1)
some of the possible things you can do with that coaster type

Realsteel89 commented:
@Gymnasiast found some bug that is possible to choose this combination 
![grafik](https://github.com/user-attachments/assets/0820fab7-f1c1-45b9-b9f9-e152c9237870)
but its not possible to build an 1 tile banked 90degree curve maybe its some bug in that construction code too

Gymnasiast commented:
It is a bug yes. No other coaster has both tiny turns and banked turns, so there is no code to disable the "tiny turn" buttons. (Note that the "tiny right turn" is enabled too.)

karst commented on distribution/changelog.txt:
```diff
@@ -1,6 +1,7 @@
 0.4.16 (in development)
 ------------------------------------------------------------------------
 - Improved: [#23010] Make AppImage compatible with Ubuntu 22.04 and Debian Bookworm again.
+- Improved: [#20587] Adding new Maurer Spinning Roller Coaster Track.
```
Move this up a line.

Realsteel89 commented:
Need new review again

SpartanFrederic104 commented:
Keep open please. This is still being worked on.

ZehMatt commented:
Lets keep it in draft until its ready then the bot will also stop nagging.

karst commented:
This'll be a perfect match with the other intamin track right? Then you could connect all four track styles together to make it super realistic, I am sure NE would be super happy with that XD

Gymnasiast commented:
@karst Could you elaborate? I’m missing the context here, I think.

karst commented:
So the Mini Coaster, the Giga Coaster and the Hydraulic launch coaster tracks (to keep it easy) perfectly match onto eachother. The mini coaster uses a zigzag spine, the giga coaster uses that times three, and the hydraulic thatone times four.
This track would be the same as the mini without the zigzag. If it perfectly fits on that, it matches Intamin's track IRL perfectly.
Example using the modern Intamin track style:
![](https://images.coasterpedia.net/thumb/1/13/Intimidator_305_brake_run.jpg/640px-Intimidator_305_brake_run.jpg)

Realsteel89 commented:
I think I need an update on this I want to add the new diagonal flat to steep long parts too and the dive loops too but after I update vs I had some trouble now with the renderer.
- On this weekend I was not home I will activate this again if it is ready

brisingr25 commented:
So I tested for a bit without trains, here's some things I found:

The track preview on the colours window cuts incorrectly and seems to be from a previous version of the track:

![spin1](https://github.com/user-attachments/assets/94ca3d67-265a-4e33-9f57-e7c57b7ea821)

The tiny turns can be selected when building banked:

![spin7](https://github.com/user-attachments/assets/b9937aa2-6be9-4fea-87e6-f5c579a7924d)

Currently you can't build diagonal gentle slopes (banked or unbanked) coming from flat diagonal, only from steep diagonal:

![spin3](https://github.com/user-attachments/assets/d1078ff3-b81d-4eee-a2ae-18a28b1f2052)

Tunnels, my beloved. The inline twist tunnel is wrong, and the tunnels for the boosters and the spin toggle control piece should be round also:

![spin6](https://github.com/user-attachments/assets/87eb2db6-9a6f-45e0-b553-6614273ad23b)

Speaking of the spinning control toggle piece, its track looks different:

![spin8](https://github.com/user-attachments/assets/d4186be5-8946-4393-90fd-1031c01cb217)

Nitpicking, but these connections on the vertical pieces could be improved:

![spin5](https://github.com/user-attachments/assets/5d9d9c1a-5fdd-4ad8-ab7d-47c41c3780be)

Weird support top on some of the large zero-g-roll pieces:

![spin2](https://github.com/user-attachments/assets/6e46bcbf-dc13-4095-b226-5e750d42e5e4)

Not a fan of how these supports change where they connect to the track depending on the angle rotation from which they're seen. Here they seem to be on the edge of the track, but if you rotate the view they seem to connect in the middle of the track:

![spin4](https://github.com/user-attachments/assets/c1ff0219-362c-4950-ac56-cd99e91fa7b1)

There's also several cases of a gap where the support meets a piece, but I think this is enough for now. Overall it looks promising! I must say that for me, the track mostly still looks a bit "pixelated" for my taste, and not quite to vanilla standard. But I've tried to change the track into the giga coaster and the mini coaster track and it connects well (I'll do some more testing on this), and I can see why the track is the shape that it is and why maybe it can't be improved or changed much. Maybe a pixel here and there manually added could be enough. I'm being nitpicky and also I lack the vocabulary or knowledge to explain what I mean properly, and I really would like to see this track in the game, but I don't think it's ready yet.

I'll do some more testing when I can, with trains running and such. Good job!
